---
title: "CD91_sarcoma_explore.Rmd"
author: "Teddy"
date: "06/09/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## CD91 Sarcoma Exploration
This code downloads the TCGA Sarcoma datasets from the cBioportal and looks at expression data related to the CD91 gene. The data will be split apart in multiple ways to compare disease status and related metrics to the level of CD91 mRNA expression.


### Set up environment
Here I load the relevant packages and set environment variables. Specialty package `cgdsr` is loaded from the [cBioportal](http://www.cbioportal.org/) website. 
```{r set_envir, echo=FALSE, warning=FALSE, message=FALSE, error=FALSE}
#'[Loading_package:]
require(pacman)
p_load(data.table, ggplot2, cgdsr)

#'[Set_variables:]
#' @none to be declared
```


### Download the datasets
Use the cBio package to download TCGA PanCancer Atlas sarcoma cancer data from [here](http://www.cbioportal.org/study/summary?id=sarc_tcga_pan_can_atlas_2018).
```{r harvest_data, echo = FALSE, warning = FALSE, message = FALSE, error = FALSE}
#' @create the CGDS object
mycgds = CGDS("https://www.cbioportal.org/")

test(mycgds)

#' @extract cancer studies list and find sarcoma-related items
studies_list <- data.table(getCancerStudies(mycgds))
sarc_studs <- studies_list[grepl(pattern = "sarc", x = cancer_study_id)]

#' @determine the study ID for the sarcoma studies we want
target_study <- sarc_studs[cancer_study_id == "sarc_tcga_pan_can_atlas_2018"]
target_study_id <- paste(target_study[1,1])

#' @get caselist (collection of samples) for our 
sarc_caselist <- getCaseLists(mycgds, target_study_id)
sarc_caselist_id <- paste(sarc_caselist[1,1])
```


### Pull Sarcoma Samples
Use the downloaded data to find the TCGA sarcoma data and isolate the RNA-seq expression data for all samples with such data.
```{r pull_samples}

#' @get available genetic profiles strings
sarc_genetic_profiles_RNAseq <- getGeneticProfiles(mycgds, target_study_id)[2,1]
sarc_genetic_profiles_muts <- getGeneticProfiles(mycgds, target_study_id)[5,1]



#' @get data slices for a specified list of genes, genetic profile and case list
RNAseq_profile_data <- getProfileData.CGDS(x = mycgds, 
                                           genes = c('LRP1'), 
                                           geneticProfiles = sarc_genetic_profiles_RNAseq, 
                                           caseList = sarc_caselist_id)


MUT_profile_data <- getProfileData.CGDS(x = mycgds, 
                                        genes = c('LRP1'), 
                                        geneticProfiles = sarc_genetic_profiles_muts, 
                                        caseList = sarc_caselist_id)


#' @get clinical data for the target sarcoma case list
sarc_clinical_data <- getClinicalData.CGDS(x = mycgds, caseList = sarc_caselist_id)

#' @get available mutation data for the sarcoma cancer study
sarc_mutation_data <- getMutationData.CGDS(x = mycgds, 
                                           caseList = sarc_caselist_id, 
                                           geneticProfile = sarc_genetic_profiles_muts, 
                                           genes = 'LRP1')
```
### Create combined set
We have pulled all of the expression data, mutation data, and clinical data for all out samples in the target sarcoma studies. Now we need to combine them into a master sheet
```{r combino}
#'[format the data]
#' @create case_id column and convert to data.table
RNAseq_profile_data$case_id <- row.names(x = RNAseq_profile_data)
RNAseq_profile_data <- data.table(RNAseq_profile_data)
setnames(x = RNAseq_profile_data, old = "LRP1", new = "LRP1_mrna_TPM")

#' @create case_id column and convert to data.table
# MUT_profile_data$case_id <- row.names(x = MUT_profile_data)
# MUT_profile_data <- data.table(MUT_profile_data)
# setnames(x = MUT_profile_data, old = "LRP1", new = "mutation")
#' [not needed]

#' @format data and remove non-important rows
sarc_clinical_data$case_id <- row.names(x = sarc_clinical_data)
sarc_clinical_data <- data.table(sarc_clinical_data)
sarc_clinical_data <- sarc_clinical_data[,.(case_id, AGE, ANEUPLOIDY_SCORE, CANCER_TYPE, 
                                            CANCER_TYPE_ACRONYM, CANCER_TYPE_DETAILED, 
                                            DAYS_LAST_FOLLOWUP, DFS_MONTHS, DFS_STATUS, 
                                            DSS_MONTHS, DSS_STATUS, ETHNICITY, 
                                            FRACTION_GENOME_ALTERED, MUTATION_COUNT,
                                            NEW_TUMOR_EVENT_AFTER_INITIAL_TREATMENT,
                                            OS_MONTHS, OS_STATUS,
                                            PFS_MONTHS, PFS_STATUS, RACE, SEX, 
                                            TUMOR_TISSUE_SITE, TUMOR_TYPE)]
#' @format the mutation data
sarc_mutation_data <- data.table(sarc_mutation_data)
sarc_mutation_data <- sarc_mutation_data[,.(case_id, mutation_type, amino_acid_change)]

#' [Merge the datasets into one]
#' @merge RNAseq and MUT profile
full_data <- merge(RNAseq_profile_data, sarc_mutation_data, by = "case_id", all.x = TRUE)
full_data <- merge(full_data, sarc_clinical_data, by = "case_id", all.x = TRUE)
```


### Visual analysis
systematically visualize all aspects of data.
```{r viz}

```